packages/create-project/src/index.ts

```diff
@@ -1,151 +1,181 @@
-  private: true,
-  scripts: {
-    build: 'tsc --build',
-    'build.watch': 'tsc --build --watch',
-    start: 'ts-node src/worker.ts',
-    'start.watch': 'nodemon src/worker.ts',
-    workflow: 'ts-node src/exec-workflow.ts',
-  },
-  devDependencies: {
-    typescript: `^${typescriptVersion}`,
-    [`@tsconfig/node${nodeMajorVersion}`]: '^1.0.0',
-    'ts-node': '^10.2.1',
-    nodemon: '^2.0.12',
-  },
-  nodemonConfig: {
-    watch: ['src'],
-    ext: 'ts',
-    execMap: {
-      ts: 'ts-node',
-    },
-  },
-};
-
-const tsConfig = {
-  extends: `@tsconfig/node${nodeMajorVersion}/tsconfig.json`,
-  version: typescriptVersion,
-  compilerOptions: {
-    emitDecoratorMetadata: false,
-    experimentalDecorators: false,
-    declaration: true,
-    declarationMap: true,
-    sourceMap: true,
-    composite: true,
-    rootDir: './src',
-    outDir: './lib',
-  },
-  include: ['src/**/*.ts'],
-  exclude: ['node_modules'],
-};
-
-/**
- * Copy sample from `source` to `target` stripping away snipsync comments
- */
-async function copySample(source: string, target: string) {
-  const code = await readFile(source, 'utf8');
-  const stripped = code.replace(/.*@@@SNIP(START|END).*\n/gm, '');
-  await writeFile(target, stripped);
-}
+import prompts from 'prompts';
+import checkForUpdate from 'update-check';
+import { createApp } from './create-project';
+import { shouldUseYarn } from './helpers/should-use-yarn';
+import { validateNpmName } from './helpers/validate-pkg';
+import packageJson from '../package.json';
 
-async function writePrettyJson(path: string, obj: any) {
-  await writeFile(path, JSON.stringify(obj, null, 2) + os.EOL);
-}
+let projectPath = '';
 
-class UsageError extends Error {
-  public readonly name: string = 'UsageError';
-}
+const program = new Commander.Command(packageJson.name)
+  .version(packageJson.version, '-v, --version')
+  .arguments('<project-directory>')
+  .usage(`${chalk.green('<project-directory>')} [options]`)
+  .action((name) => {
+    projectPath = name;
+  })
+  .option(
+    '--use-yarn',
+    `
 
-interface Template {
-  copySources(sampleDir: string, targetDir: string): Promise<void>;
-}
+  Use yarn instead of npm
+`
+  )
+  .option(
+    '-e, --example [name]|[github-url]',
+    `
+
+  An example to bootstrap the app with. You can use an example name
+  from https://github.com/temporalio/samples-node or a GitHub URL. 
+  The URL can use any branch and/or subdirectory
+`
+  )
+  .option(
+    '--example-path <path-to-example>',
+    `
+
+  In a rare case, your GitHub URL might contain a branch name with
+  a slash (e.g. bug/fix-1) and the path to the example (e.g. foo/bar).
+  In this case, you must specify the path to the example separately:
+  --example-path foo/bar
+`
+  )
+  .allowUnknownOption()
+  .parse(process.argv);
+
+async function run(): Promise<void> {
+  if (typeof projectPath === 'string') {
+    projectPath = projectPath.trim();
+  }
+
+  if (!projectPath) {
+    const res = await prompts({
```

Comment:
Interesting choice going interactive, I guess some devs prefer that.

Reply:
It feels nice to me when tools have prompts, and a lot of the modern/popular tools I've used do. It's used only as a fallback here when command-line args aren't provided.

Reply:
👍